File: scripts/migrate-product-cards.js
// Script to migrate all articles from {{< amazon >}} to {{< product-card >}}
const fs = require('fs');
const path = require('path');
const POSTS_DIR = path.join(process.env.HOME, 'GIT/PP/mombabypicks/content/posts');

const files = fs.readdirSync(POSTS_DIR).filter(f => f.endsWith('.md'));

let total = 0;
let articles_updated = 0;

for (const file of files) {
  const fp = path.join(POSTS_DIR, file);
  let content = fs.readFileSync(fp, 'utf8');
  const original = content;
  
  // Find all {{< amazon >}} shortcodes
  const amazonRegex = /{{<\s*amazon\s+url="https:\/\/www\.amazon\.com\/(?:[^"]*\/)?dp\/([A-Z0-9]{10})(?:\?[^"]*)?"\s+text="[^"]*"\s*>}}/g;
  let match;
  let changed = false;
  
  while ((match = amazonRegex.exec(content)) !== null) {
    const asin = match[1];
    const fullMatch = match[0];
    const pos = match.index;
    
    // Find the nearest product heading (### N. Product Name — Badge) before this amazon link
    // Also try to get title and badge from nearby context
    const beforeMatch = content.substring(0, pos);
    const headingMatch = beforeMatch.match(/###\s+\d+\.\s+(.+?)(?:\s*[—–-]\s*(.+))?[\s\S]*?(?=###|\n\n|$)/);
    // Actually, simpler approach: just get the nearest ### heading
    const simpleHeading = beforeMatch.match(/###\s+\d+\.\s+([^\n]+)/);
    
    let title = asin;
    let badge = '';
    
    if (simpleHeading) {
      const headingText = simpleHeading[1].trim();
      // Extract badge if present (text after — or – or -)
      const badgeSep = headingText.match(/\s*[—–-]\s*(.+)/);
      if (badgeSep) {
        badge = badgeSep[1].trim();
        title = headingText.substring(0, headingText.indexOf(badgeSep[0])).trim();
      } else {
        title = headingText;
      }
    }
    
    // If no badge from heading, try pick-cards
    if (!badge) {
      const pickMatch = beforeMatch.match(new RegExp('card' + '\\d+' + 'label="([^"]*)"[\\s\\S]*?' + asin, 'i'));
      // simpler: look for any card label
      const labelMatch = beforeMatch.match(/card\d+label="([^"]+)"/g);
      if (labelMatch) {
        // Get the last one before position
        const lastLabel = labelMatch[labelMatch.length - 1];
        const lbl = lastLabel.match(/card\d+label="([^"]+)"/);
        if (lbl) badge = lbl[1];
      }
    }
    
    // Get price from the product section (look for "**Price:**" before this point but after the heading)
    const priceMatch = beforeMatch.match(/\*\*Price:\*\*\s*\$?([\d.]+)/);
    const price = priceMatch ? '$' + priceMatch[1] : '';
    
    // Clean title - remove anything after "—" since that's the badge
    const cleanTitle = title.replace(/\s*—\s*.*$/, '').trim();
    
    const newShortcode = `{{< product-card url="https://www.amazon.com/dp/${asin}?tag=mombabypick00-20" img="https://images-na.ssl-images-amazon.com/images/P/${asin}.01.L.jpg" title="${cleanTitle}"${badge ? ` badge="${badge}"` : ''}${price ? ` price="${price}"` : ''} >}}`;
    
    content = content.replace(fullMatch, newShortcode);
    changed = true;
    total++;
  }
  
  if (changed) {
    fs.writeFileSync(fp, content);
    articles_updated++;
    process.stdout.write('📝');
  }
}

console.log(`\n✅ ${articles_updated} articles updated, ${total} total amazon links migrated`);

File: scripts/migrate-product-cards.py
import re, os

POSTS = os.path.expanduser('~/GIT/PP/mombabypicks/content/posts')

total = 0
updated = 0

# Patterns for amazon links
SHORTCODE = r'\{\{<\s*amazon\s+url="https?://[^"]*?/dp/([A-Z0-9]{10})(?:\?[^"]*)?"\s+text="[^"]*"\s*>\}\}'
MARKDOWN = r'\[Check[^\]]*\]\(https?://(?:www\.)?amazon\.com(?:/[^/]+)?/dp/([A-Z0-9]{10})[^)]*\)'

for fname in sorted(os.listdir(POSTS)):
    if not fname.endswith('.md'):
        continue
    fp = os.path.join(POSTS, fname)
    with open(fp) as f:
        content = f.read()
    orig = content
    
    # Get all matches with positions
    items = []
    for m in re.finditer(SHORTCODE, content):
        items.append((m.start(), m.end(), m.group(0), m.group(1)))
    for m in re.finditer(MARKDOWN, content):
        items.append((m.start(), m.end(), m.group(0), m.group(1)))
    
    items.sort(key=lambda x: -x[0])  # Reverse order for in-place replacement
    
    if not items:
        continue
    
    for start, end, full_match, asin in items:
        # Find last product heading before this position
        before = content[:end]
        h = re.findall(r'#{1,3}\s+(?:\d+\.\s+)?([^\n]+)', before)
        
        title = asin
        badge = ''
        price = ''
        
        if h:
            ht = h[-1].strip()
            parts = re.split(r'\s*[—–]\s*', ht, maxsplit=1)
            title = parts[0].strip()
            if len(parts) > 1:
                badge = parts[1].strip()
        
        # Get price from section between heading and amazon link
        section = before
        if h:
            last_idx = before.rfind('###')
            if last_idx == -1:
                last_idx = before.rfind('## ')
            section = before[last_idx:] if last_idx >= 0 else before
        
        pm = re.search(r'\*\*Price:\*\*\s*\$?([\d.]+)', section)
        if pm:
            price = '$' + pm.group(1)
        
        # Clean title
        title = title.split(' — ')[0].split(' – ')[0].split(' - ')[0].strip()
        title = re.sub(r'\s+', ' ', title)
        
        new = f'{{{{< product-card url="https://www.amazon.com/dp/{asin}?tag=mombabypick00-20" img="https://images-na.ssl-images-amazon.com/images/P/{asin}.01.L.jpg" title="{title}"'
        if badge:
            new += f' badge="{badge}"'
        if price:
            new += f' price="{price}"'
        new += ' >}}'
        
        content = content[:start] + new + content[end:]
        total += 1
    
    if content != orig:
        with open(fp, 'w') as f:
            f.write(content)
        updated += 1
        print('📝', fname)

print(f'\n✅ {updated} articles updated, {total} links migrated')
